extensions/2.0/Vendor/GRIFFEL_bim_data/README.md

```diff
@@ -0,0 +1,206 @@
+└Storey 2
+ ├Category 1
+ └Category 2
+  ├Element 1
+  └Element 2
+```
+Where:
+Model - is a building,
+Storey - includes all elements which lay or rise up starting from a particular level,
+Category - groups elements by some parameter (columns, furniture, ducts, etc.),
+Element - leaf `node` with `mesh` which represents a real construction product.
+
+### Approach
+A single piece of metadata for the node is represented by `property`. Property is a key-value pair, like 'name of the object's parameter' - 'its value'. To keep it simple name and value of the property are `strings`.
+Node can also have a `type`. Type is a set of properties which is common for many nodes.
+#### Example:
+There are two timber doors in the building. Each is 900 mm (3') wide. First is 2,1 m (7') high, second is 2,4 m (8') high.
+In this example properties are: "Width - 900 mm", "Height - 2,1 m", "Height - 2,4 m", "Material - Timber". 
+A single `type` here is represented by the collection consisting of the first and the last properties as width and material are common properties for both doors.
+
+To save some space for storing properties they are written only once in the properties collection and referenced by each node where needed. To save even more space each unique property name and each unique property value are stored in separate collections. This approach is beneficial over keeping metadata in `extras` for large amounts of data.
+
+Types and nodes reference properties by index.
+Each node can reference any number of properties and only one type.
+
+## Implementation notice
+
+Given the same two doors from the example above, there are two options to implement the extension:
+* ### Embed properties into glTF json
+```javascript
+"nodes": [
+  {
+    "name": "Door 1",
+    "extensions": {
+      "GRIFFEL_bim_data": {
+        "properties": [
+          0
+        ],
+        "type": 0
+      }
+    }
+  },
+  {
+    "name": "Door 2",
+    "extensions": {
+      "GRIFFEL_bim_data": {
+        "properties": [
+          3
+        ],
+        "type": 0
+      }
+    }
+  }
+],
+"extensions": [
+  "GRIFFEL_bim_data": {
+    "propertyNames": [
+      "Height",
+      "Width",
+      "Material"
+    ],
+    "propertyValues": [
+      "2,1 m",
+      "900 mm",
+      "Timber"
+      "2,4 m",
+    ],
+    "properties": [
+      {
+        "name": 0,
+        "value": 0
+      },
+      {
+        "name": 1,
+        "value": 1
+      },
+      {
+        "name": 2,
+        "value": 2
+      },
+      {
+        "name": 0,
+        "value": 3
+      }
+    ],
+    "types": [
+      {
+        "properties": [
+          1,
+          2
+        ]
+      }
+    ]
+  }
+]
+```
+
+* ### Write properties to separate binary file
+
+In this case each node points to the buffer where metadata for this node can be found. File is a serialized 'GRIFFEL_bim_data' extension object with additional `nodeProperties` field. This field maps node properties and types to nodes by nodeId.
```

Comment:
Instead of a buffer, the extension should point to a buffer view. That would allow packing everything into GLB with the existing tools. Applications would still be able to use separate MsgPack files if needed.

Reply:
Hm, good idea, but in that case which `target` the buffer view should have?
Also it's perfectly OK not to pack .meta file into GLB - as the benefit of .meta file is that it can go separately. Extension users can share just gltf file first and than, if needed additionally share/sent/load/read .meta file.
Also if GLB is needed entire extension can be written into gltf in json format, as stated in "Embed properties into glTF json" paragraph.

Reply:
`bufferView.target` is used only for mesh data such as vertex attributes and indices. For images and animations, that property remains undefined.

In general, using buffer views gives extra flexibility to users and applications as they can choose per-asset what to embed and what to store separately:
- `GRIFFEL_bim_data.buffer` -> `buffer` -> independent buffer is always enforced;

alternatively
- `GRIFFEL_bim_data.bufferView` -> `bufferView` -> `buffer` -> may use the same external buffer as other data (implicitly enabling GLB), may use independent files as before.

Reply:
Done.